doc/swish/pregexp.tex

```diff
@@ -0,0 +1,817 @@
+the preceding subpattern.  \var{m} must be a nonnegative integer.
+
+The quantifier \code{\{\var{m},\var{n}\}} matches at least \var{m} and
+at most \var{n} instances.  \var{m} and \var{n} are nonnegative
+integers with $\var{m} \le \var{n}$.  You may omit either or both
+numbers, in which case \var{m} defaults to 0 and \var{n} to infinity.
+
+It is evident that \code{+} and \code{?} are abbreviations for
+\code{\{1,\}} and \code{\{0,1\}} respectively.  \code{*} abbreviates
+\code{\{,\}}, which is the same as \code{\{0,\}}.
+
+\code{(pregexp-match "[aeiou]\{3\}" "vacuous")} $\Rightarrow$
+\code{("uou")}
+
+\code{(pregexp-match "[aeiou]\{3\}" "evolve")} $\Rightarrow$ \code{\#f}
+
+\code{(pregexp-match "[aeiou]\{2,3\}" "evolve")} $\Rightarrow$
+\code{\#f}
+
+\code{(pregexp-match "[aeiou]\{2,3\}" "zeugma")} $\Rightarrow$
+\code{("eu")}
+
+\subsubsection {Non-greedy Quantifiers}
+
+The quantifiers described above are \emph{greedy}, i.e., they match
+the maximal number of instances that would still lead to an overall
+match for the full pattern.
+
+\code{(pregexp-match "<.*>" "<tag1> <tag2> <tag3>")} $\Rightarrow$
+\code{("<tag1> <tag2> <tag3>")}
+
+To make these quantifiers \emph{non-greedy}, append a \code{?} to
+them.  Non-greedy quantifiers match the minimal number of instances
+needed to ensure an overall match.
+
+\code{(pregexp-match "<.*?>" "<tag1> <tag2> <tag3>")} $\Rightarrow$
+\code{("<tag1>")}
+
+The non-greedy quantifiers are respectively: \code{*?}, \code{+?},
+\code{??}, \code{\{\var{m}\}?}, and \code{\{\var{m},\var{n}\}?}.  Note
+the two uses of the metacharacter \code{?}.
+
+\subsection {Clusters}
+
+\emph{Clustering}, i.e., enclosure within parentheses
+\code{(}\ldots\code{)}, identifies the enclosed subpattern as a single
+entity.  It causes the matcher to \emph{capture} the \emph{submatch},
+or the portion of the string matching the subpattern, in addition to
+the overall match.
+
+\code{(pregexp-match "([a-z]+) ([0-9]+), ([0-9]+)" "jan 1, 1970")} \\
+$\Rightarrow$ \code{("jan 1, 1970" "jan" "1" "1970")}
+
+Clustering also causes a following quantifier to treat the entire
+enclosed subpattern as an entity.
+
+\code{(pregexp-match "(poo )*" "poo poo platter")} $\Rightarrow$
+\code{("poo poo " "poo ")}
+
+The number of submatches returned is always equal to the number of
+subpatterns specified in the regexp, even if a particular subpattern
+happens to match more than one substring or no substring at all.
+
+\code{(pregexp-match "([a-z ]+;)*" "lather; rinse; repeat;")} \\
+$\Rightarrow$ \code{("lather; rinse; repeat;" " repeat;")}
+
+Here the \code{*}-quantified subpattern matches three times, but it is
+the last submatch that is returned.
+
+It is also possible for a quantified subpattern to fail to match, even
+if the overall pattern matches.  In such cases, the failing submatch
+is represented by \code{\#f}.
+
+\begin{alltt}
+(define date-re
+  ;; match ‘month year’ or ‘month day, year’.
```

Comment:
The quotes here render strangely for me. `;; match âĂŸmonth yearâĂŹ or âĂŸmonth day, yearâĂŹ.`

Reply:
Got 'em, thanks!